PenLineModel.cpp

```diff
@@ -5,9 +5,39 @@
 //Our includes
 #include "PenLineModel.h"
 
+class PenLineModelUndoCommand : public QUndoCommand {
+public:
+    PenLineModelUndoCommand(PenLineModel *model) : QUndoCommand("Draw Line", nullptr), m_penLineModel(model), m_oldLines(model->m_lines) {}
+
+    void finish()
```

Comment:
This likely can be removed. When the command is pushed it will execute redo and set newlines.

Reply:
Interesting -- the previous Undo frameworks I've used didn't call `redo` upon pushing, so I didn't notice this behavior.